diff --git a/lib/roles/h2/hpack.c b/lib/roles/h2/hpack.c
index 68629e6f..6ef628b8 100644
--- a/lib/roles/h2/hpack.c
+++ b/lib/roles/h2/hpack.c
@@ -1110,7 +1110,7 @@ pre_data:
 
 		for (n = 0; n < 8; n++) {
 			if (h2n->huff) {
-				char b = (c >> 7) & 1;
+			        char b = (char)((c >> 7) & 1);
 				prev = h2n->hpack_pos;
 				h2n->hpack_pos = (uint16_t)huftable_decode(
 						(int)h2n->hpack_pos, b);
diff --git a/lib/core/lws_map.c b/lib/core/lws_map.c
index d149d86752..b319d79f49 100644
--- a/lib/core/lws_map.c
+++ b/lib/core/lws_map.c
@@ -29,11 +29,11 @@ typedef struct lws_map_hashtable {
 	lws_dll2_owner_t		ho;
 } lws_map_hashtable_t;
 
-typedef struct lws_map {
+struct lws_map {
 	lws_map_info_t			info;
 
 	/* array of info.modulo x lws_map_hashtable_t overallocated */
-} lws_map_t;
+};
 
 typedef struct lws_map_item {
 	lws_dll2_t			list; /* owned by hashtable */

